app/src/main/java/org/mozilla/focus/activity/MainActivity.java

```diff
@@ -283,6 +284,10 @@ public void onValueChanged(@NonNull List<Session> sessions) {
                     // We have at least one session. Show a fragment for the current session.
                     showBrowserScreenForCurrentSession();
                 }
+
+                if (Settings.getInstance(MainActivity.this).shouldShowFirstRun()) {
```

Comment:
Hm, should this need to be in the session manager callback? I think you could just put it at the end of onCreate (assuming all the views are created by then).

Reply:
Yeah – we initialize the sessions and set the screen according to the observer, e.g. `showHomeScreen` and `showBrowserScreenForCurrentSession` are also in here.